enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/analytics/DefaultAnalyticsEventPublisher.java

```diff
@@ -94,23 +94,24 @@ public void handleGRPCLogMsg(StreamAccessLogsMessage message) {
 
     @Override
     public void handleWebsocketFrameRequest(WebSocketFrameRequest webSocketFrameRequest) {
-        AnalyticsDataProvider  provider = new ChoreoAnalyticsForWSProvider(webSocketFrameRequest);
+        AnalyticsDataProvider provider = new ChoreoAnalyticsForWSProvider(webSocketFrameRequest);
         collectDataToPublish(provider);
     }
 
     @Override
     public void init(Map<String, String> configuration) {
-        boolean elkEnabled = org.wso2.choreo.connect.enforcer.analytics.AnalyticsConstants.ELK_TYPE
-                .equalsIgnoreCase(ConfigHolder.getInstance().getConfig().getAnalyticsConfig().getType());
-        if (!elkEnabled && (StringUtils.isEmpty(configuration.get(AnalyticsConstants.AUTH_URL_CONFIG_KEY))
-                || StringUtils.isEmpty(AnalyticsConstants.AUTH_TOKEN_CONFIG_KEY))) {
-            logger.error(AnalyticsConstants.AUTH_URL_CONFIG_KEY + " and / or " +
-                    AnalyticsConstants.AUTH_TOKEN_CONFIG_KEY +
-                    "  are not provided. Hence assigning default values");
-            configuration.put(AnalyticsConstants.AUTH_TOKEN_CONFIG_KEY, "");
-            configuration.put(AnalyticsConstants.AUTH_URL_CONFIG_KEY, "https://localhost:8080");
-            return;
+
+        if (StringUtils.isEmpty(configuration.get(PUBLISHER_REPORTER_CLASS_CONFIG_KEY))) {
```

Comment:
```suggestion
        if (StringUtils.isEmpty(configuration.get(PUBLISHER_REPORTER_CLASS_CONFIG_KEY)) || 
            "org.wso2.am.analytics.publisher.reporter.cloud.DefaultAnalyticsMetricReporter".equals(configuration.get(PUBLISHER_REPORTER_CLASS_CONFIG_KEY) {
```

Reply:
Just for handling an edge case, we could even proceed without this change considering the likelihood of the occurance.

Reply:
Yeah, in the general case, we don't need to use this.